src/Elastic.Apm/Metrics/MetricsProvider/SystemTotalCpuProvider.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Elastic.Apm.Api;
+
+namespace Elastic.Apm.Metrics.MetricsProvider
+{
+	internal class SystemTotalCpuProvider : IMetricsProvider
+	{
+		private const string SystemCpuTotalPct = "system.cpu.total.norm.pct";
+
+		public SystemTotalCpuProvider()
+		{
+			if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) return;
+
+			var procStatValues = ReadProcStat();
+			if (!procStatValues.success) return;
+
+			_prevIdleTime = procStatValues.idle;
+			_prevTotalTime = procStatValues.total;
+		}
+
+		private double _prevIdleTime;
+		private double _prevTotalTime;
+		private PerformanceCounter _processorTimePerfCounter;
+
+		public int ConsecutiveNumberOfFailedReads { get; set; }
+		public string DbgName => "total system CPU time";
+
+		private static (bool success, double idle, int total) ReadProcStat()
```

Comment:
Any reason why `idle` is not `int` as well? I would use `long` for both (and for `numbers` array) because if the units are USER_HZ or Jiffies (typically hundredths of a second) (http://www.linuxhowtos.org/System/procstat.htm) then it will not fit into `int` after 8 months.

Reply:
I saw some sample code, and most of them used `int`, those data types are probably leftovers from some previous changes. I changed everything to `long`.